packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -42,10 +42,31 @@ enum FlutterPluginVersion {
 // Investigation documented in #13975 suggests the filter should be a subset
 // of the impact of -q, but users insist they see the error message sometimes
 // anyway.  If we can prove it really is impossible, delete the filter.
+// This technically matches everything *except* the NDK message, since it's
+// passed to a function that filters out all lines that don't match a filter.
 final RegExp ndkMessageFilter = RegExp(r'^(?!NDK is missing a ".*" directory'
   r'|If you are not using NDK, unset the NDK variable from ANDROID_NDK_HOME or local.properties to remove this warning'
   r'|If you are using NDK, verify the ndk.dir is set to a valid NDK directory.  It is currently set to .*)');
 
+// This regex is intentionally broad. AndroidX errors can manifest in multiple
```

Comment:
Should we be overly broad, or overly conservative here?  Do we know of any particular cases we can mis-identify as an AndroidX issue?

Reply:
At this point I think it's better to direct people to a help page that doesn't apply to them than to miss people who are completely blocked on this. The gradle messages here don't give enough information to debug this on their own. You can click through the issue links for some examples of what they end up looking like. I am only logging when Gradle exits with a failing status code to try and minimize the logs to when there's actually a crash.

Out of the super broad nets I'm throwing here:

* "androidx" will only be mentioned by Gradle if it's being brought into the project by something in the first place. It's highly likely that it's related. It could only be unrelated if the user already had brought it in and then had an error in Java files that use it directly.
* "android.support" could be mentioned as an unrelated error if it's being manipulated somewhere in Java code, like a broken import or calling a nonexistent method. So if users are manipulating Java directly to use the old support libraries (or their dependencies are) and the code breaks based on that this warning will get thrown up as a false positive. I think it's possible but probably unlikely, and when this happens it's likely that users will know that it's a false positive since they would probably have just been editing the Java files. (Unless they brought in a plugin that shipped in a broken state, but that also seems unlikely.)
* "AAPT" is really broad. It's just a reference to [the step where resources are compiled](https://developer.android.com/studio/command-line/aapt2). I don't know all the things that could cause this but I suspect that there's a lot of ways to get into a broken state here besides the AndroidX bug. Looking at our repo there's a large amount of closed issues that mention this in the past. If I look at open gradle issues in our repo flutter/flutter#2464 is open for it still. Unfortunately this is also the main way this problem manifests right now.

Reply:
Appreciate the explanation.